build_tools/github/trigger_hosting.sh

```diff
@@ -31,10 +31,29 @@ else
      BRANCH=$HEAD_BRANCH
 fi
 
-curl --request POST \
+# Circle CI REST API might return HTTP response with 202 status code
+# even when the POST requests fail.
+CIRCLE_CI_RESPONSE=$(curl --request POST \
      --url https://circleci.com/api/v2/project/gh/$GITHUB_REPOSITORY/pipeline \
      --header "Circle-Token: $CIRCLE_CI_TOKEN" \
      --header "content-type: application/json" \
      --header "x-attribution-actor-id: github_actions" \
      --header "x-attribution-login: github_actions" \
-     --data \{\"branch\":\"$BRANCH\",\"parameters\":\{\"GITHUB_RUN_URL\":\"$GITHUB_RUN_URL\"\}\}
+     --data \{\"branch\":\"$BRANCH\",\"parameters\":\{\"GITHUB_RUN_URL\":\"$GITHUB_RUN_URL\"\}\})
+
+echo CIRCLE_CI_RESPONSE
+
+CIRCLE_CI_RESPONSE_MESSAGE=$(cat $CIRCLE_CI_RESPONSE | jq ".message")
+
+if [[ "$CIRCLE_CI_RESPONSE_MESSAGE" =~ .*"Not Found".* ]]; then
+  echo "The endpoint has not been found on Circle CI REST API. Please check the request correctness."
```

Comment:
Is this case where CircleCI responds with a status 202? The way I understand your PR description is that sometimes (at random) CircleCI responds with this status. If yes, could we make the error message more like "Sometimes CircleCI randomly responds with status 202, best strategy is to try again."

I tried to find a way to get the HTTP response status from curl but couldn't :( [Status 202](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202) seems like a weird code to see. And if it happens "at random" then it is probably related to something being broken/not quite right on their side. So there is nothing to do for the user, they should just try again. Hence the suggestion to tweak the error message.

Reply:
It is not random: in case of failure Circle CI returns 20X status codes where a 4XX or 5XX status code must be returned. This situations is silencing failures but it must not.

Should the comment be rewritten so that this is clear?

Reply:
Could you explain when this happens? I don't get it :( I think my answer to "rewrite this to make it clearer?" is yes, but I don't have a suggestion because I don't understand when/why this happens.

Reply:
This happened for instance when a token was not set before #25338, see [those logs](https://github.com/scikit-learn/scikit-learn/actions/runs/3872313863/jobs/6601020506#step:3:16).

There, permissions were denied because no token was set. Yet the API returned a response of status code `20X` not making `curl(1)` errs (which is the case when faced to response of `4XX` or `5XX` status code). Thus the job was not reported as failing but instead as having successfully completed. 

Does this answer your question? :slightly_smiling_face:

Reply:
Ah, wait I have not committed the last version of the message. With cd1125c, this should now be clearer.

Reply:
LGTM